src/dmd/dsymbolsem.d

```diff
@@ -6499,3 +6486,40 @@ void aliasSemantic(AliasDeclaration ds, Scope* sc)
             ScopeDsymbol.multiplyDefined(Loc.initial, sx, ds);
     }
 }
+
+/**
+Checks if a function declaration's lifetime attributes (e.g. `scope` and `return`)
+are valid
+Params:
+    fd = The function declaration to check
+    sc = The semantic scope in which to check
+*/
+private void checkLifetimeAttributes(FuncDeclaration fd, Scope* sc)
+{
+    if (fd.type)
```

Comment:
Code is usually better if you return early. So instead of nesting the whole function in an `if`, use:
```d
if (fd.type)
    return;
```